app/goal_routes.py

```diff
@@ -0,0 +1,97 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from .routes_helper import validate_model
+
+goals_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response(jsonify({"details": "Invalid data"}), 400)
+    # Create a new goal in the database
+    new_goal = Goal(
+        title=request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body = dict(goal = new_goal.to_dict())
+
+    return jsonify(response_body), 201
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+
+    goals = Goal.query.all()
+    goal_list = [goal.to_dict() for goal in goals]
+
+
+    return jsonify(goal_list), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model(Goal,goal_id)
+    response_body = dict(goal = goal.to_dict())
+
+    return jsonify(response_body), 200
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal,goal_id)
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    response_body = dict(goal = goal.to_dict())
+
+    return jsonify(response_body) , 200
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({"details": f"Goal {goal.id} \"{goal.title}\" successfully deleted"}, '200 OK')
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def adding_task_ids(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    task_ids = request_body.get("task_ids")
```

Comment:
I like that you used .get() to get the task_ids. Consider passing a default value that is an empty list, like: `task_list = request_body.get("task_ids", [])`

As is, if the key "task_ids" is not in request_body then task_list will reference None. Then when you try to loop over None, you'll run into another unhandled exception (Nonetype is not iterable or something like that)